spec/classes/hiera_spec.rb

```diff
@@ -309,7 +309,7 @@
         it { is_expected.to contain_class('hiera::deep_merge') }
       end
     end
-  elsif Puppet.version =~ %r{^4}
+  elsif Puppet.version =~ %r{^6}
```

Comment:
I think we should do more than just this.  There's loads of dead tests that could be removed completely.  If we're dropping puppet 4, maybe there's also code that should be cleared up and/or defaults that should be more sane.

Reply:
I dropped some of the legacy tests. Should we keep it as it is and merge it? We can raise an issue to cleanup the module later. That would allow us to move forward and release the module.